.changelog/unreleased/dependencies/2346-upgrade-golangci-lint.md

```diff
@@ -0,0 +1 @@
+* golangci/golangci-lint-action` bumped to 8 (from 6) and golangci-lint v2.1.6 [#2346](https://github.com/provenance-io/provenance/issues/2346).
```

Comment:
These should be split into multiple entries. The dependencies section is a little special in our changlog, and works best if there's only one thing per line.

```suggestion
* `golangci/golangci-lint-action` bumped to 8 (from 6) [#2346](https://github.com/provenance-io/provenance/issues/2346).
* `golangci-lint` bumped to v2.1.6 (from v1.60) [#2346](https://github.com/provenance-io/provenance/issues/2346).
```